demo_modules/slideshow/slideshow.js

```diff
@@ -128,8 +128,9 @@ class ImageServer extends ModuleInterface.Server {
       // which will cause us to send the init event. Then, we might send 1 or 2
       // init messages, and we leave it up to the client to not process it
       // twice.
-      loadingComplete.then(initHandler);
-      socket.once('req_init', initHandler);
+      Promise.race(loadingComplete, new Promise(resolve => {
```

Comment:
Yes, but the new code doesn't race the promises, as this doesn't make any sense. The bug is that we were sending 'init' with the 'config' loaded from the load strategy without waiting for it to init. This would cause the clients to not always have credentials, which results in a black screen.